docs/proposals/respect-rbac-for-resource-exclusions.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Respect RBAC for Resource Inclusions/Exclusions
+
+authors:
+- "@gdsoumya"
+- "@alexmt"
+
+sponsors:
+- TBD
+
+reviewers:
+- TBD
+
+approvers:
+- TBD
+
+creation-date: 2023-05-03
+
+---
+
+# Enhancement Idea
+
+This is a proposal to provide the ability to configure argocd controller, to respect the current RBAC permissions 
+when handling resources besides the already existing resource inclusions and exclusions.
+
+## Summary
+
+Argo CD administrator will be able to configure in `argocd-cm`, whether to enable or disable(default) the feature where the controller will 
+only monitor resources that the current service account allows it to read.
+
+## Motivation
+
+Some users restrict the access of the argocd to specific resources using rbac and this feature will enable them to continue 
+using argocd without having to manually configure resource exclusions for all the resources that they don't want argocd to be managing.
+
+## Proposal 
+
+The configuration for this will be present in the `argocd-cm`, we will add new boolean field `resource.respectRBAC` in the
+cm which can be set to `true` to enable this feature, by default the feature is disabled.
+
+The feature will also modify `gitops-engine` pkg to add a `SelfSubjectAccessReview` request before adding any resource to the watch list, 
+which will make sure that argocd only monitors resources that it has access to.
```

Comment:
I would suggest not to perform an additional request to `SelfSubjectAccessReview`, because it potentially doubles the number of requests required for building up the cache. On large clusters, this is problematic already as of today.

Reply:
Instead, I'd like to propose evaluating the API response for a given resource during list and/or establishing the watch.

Reply:
Agreed, will update proposal to match the same.

Reply:
@jannfis I presented the proposal in today's meeting and there was a concern with depending on the api response, it was about false positives that might be encountered due to excessive load on the kube api server and also due to env-specific proxies. So I am adding back the `SelfSubjectAccessReview` implementation option besides the api response approach with their advantages and disadvantages as discussed in the call today.